### ISSUE TITLE
GetOptionChain - missing implied volatility?

### ISSUE THREAD

**Issue by romanlelek** [joshuaulrich/quantmod#224]
Hi, 

I have a question, why in function `getOptionChain()` is missing a column with implied volatility? For example:
```r
(option_chain_AAPL <- getOptionChain("AAPL"))
#                     Strike  Last        Chg   Bid   Ask Vol   OI
# AAPL180413C00135000  135.0 43.85  0.0000000 31.50 35.50   9   12
```
But when I have a look on web:
https://finance.yahoo.com/quote/AAPL/options?p=AAPL
I can see:

> Contract Name | Last Trade Date | Strike | Last Price | Bid | Ask | Change | % Change | Volume | Open Interest | **Implied Volatility**

Is it possible to add this column too? 

Update: I have added the column on my end v, but I think that others would be interested too.

Thank you.

Best regards,
Roman

**Comment by joshuaulrich** [comment]
I had purposefully left out the implied volatility column because it's my opinion that you should calculate it yourself rather than assume Yahoo's calculation is relevant for your situation.

Even though I still have that opinion, it probably makes more sense to just return the data as Yahoo sends it. Thoughts?

**Comment by romanlelek** [comment]
Yeah, in fact, I am calculating implied volatility by myslef, but I would like to compare it against implied volatility from Yahoo.

Today, I have discovered strange behaviour with function getOptionChain:
I would like to get optionChain for this date range - "2018-04-19/2018-07-18"

```
test1<-getOptionChain("PFE", "2018-04-19/2018-07-18")
test2<-getOptionChain("SPY", "2018-04-19/2018-07-18")

```
test1 result is as expected - data frame with various expiration dates
test2 shows Error in open.connection(con, "rb") : HTTP error 404.

Any ideas why SPY can not open connection? I have tried several tickers and only SPY doesn't work..

Best regards,
Roman

**Comment by simaan84** [comment]
Hi Josh, 

I am have been using the getOptionChain on a recurring basis. However, the package drops information regarding the last trade date. Hence, if I download the data on a daily basis, the date during the data was downloaded can be misleading. I was wondering whether it would be possible to keep track of such info? I would appreciate any feedback on this.

Cheers,
Majeed

**Comment by joshuaulrich** [comment]
Hi @romanlelek, I'm not sure why you would get a 404 error.  I wasn't able to try your code when you first encountered the issue, but your second line works now (with updated expiry dates).

Hi @simaan84, what would you think of adding an "updated" attribute to the result, similar to what `getSymbols()` does with the xts objects it returns?

**Comment by romanlelek** [comment]
Hi @joshuaulrich, 

well, I have tested it now and it works....strange...maybe Yahoo fixed something on their side. No problem anymore.

Best regards,
R.

**Comment by simaan84** [comment]
I did not get the chance to address above earlier. There seems to be an issue running when it comes to stacking the data in the data.frame
"Error in data.frame(Strike = strike, Last = lastprice, Chg = change, Bid = bid, : object 'volume' not found"

I reckon this has to do with the NewToOld function in the source file. I did some adjustment to the code. It seems to run smoothly now. In addition, I added IV and the full variables in the new edit. Let me know what you think.

**Comment by hg2581** [comment]
I am using the current version in CRAN and just ran into these two problems:
> SPY0529<-getOptionChain("SPY", NULL)
Error in data.frame(Strike = strike, Last = lastprice, Chg = change, Bid = bid,  : 
  object 'bid' not found
and
> VIX0529<-getOptionChain("^VIX", NULL)
Error in data.frame(Strike = strike, Last = lastprice, Chg = change, Bid = bid,  : 
  object 'ask' not found
Not sure what the reason might be?

**Comment by joshuaulrich** [comment]
@hg2581 I wasn't able to replicate that error, so I assume it was a temporary issue with Yahoo.

I just noticed that #304 was a duplicate of this one, and it has been closed as fixed.  So I'm going to close this one too, since it is no longer an issue.

**Comment by joshuaulrich** [comment]
> There seems to be an issue running when it comes to stacking the data in the data.frame
> "Error in data.frame(Strike = strike, Last = lastprice, Chg = change, Bid = bid, : object 'volume' not found"

@simaan84, this was reported in #299 and fixed in #300, both by @jrburl.

**Comment by joshuaulrich** [comment]
@hg2581 the 'bid' and 'ask' errors were fixed by @jrburl in #302.
